apps/passport-server/src/database/queries/frogcrypto.ts

```diff
@@ -248,26 +248,89 @@ export async function incrementScore(
   return res.rows[0];
 }
 
+/**
+ * Common table expressions for scoreboard queries.
+ *
+ * Postgres will optimize the query by pushing the where clause into the CTEs.
+ */
+const ScoreCTEs = `
+with tg as (
+  select distinct on (semaphore_id)
+  telegram_username,
+  semaphore_id
+  from telegram_bot_conversations
+  order by semaphore_id, id desc
+),
+scores as (
+  select
+  semaphore_id,
+  '0x' || encode(sha256('frogcrypto_' || semaphore_id::bytea), 'hex') as semaphore_id_hash,
```

Comment:
idk about this performance-wise, just flagging it for you to think about.

Reply:
seems fine especially we are limiting the length of leaderboard